flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/rules/logical/FlinkLimit0RemoveRule.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.planner.plan.rules.logical;
+
+import org.apache.calcite.plan.RelOptRuleCall;
+import org.apache.calcite.plan.RelRule;
+import org.apache.calcite.rel.RelNode;
+import org.apache.calcite.rel.core.Sort;
+import org.apache.calcite.rex.RexLiteral;
+import org.immutables.value.Value;
+
+/** Planner rule that rewrites `limit 0` to empty [[org.apache.calcite.rel.core.Values]]. */
+@Value.Enclosing
+public class FlinkLimit0RemoveRule
+        extends RelRule<FlinkLimit0RemoveRule.FlinkLimit0RemoveRuleConfig> {
```

Comment:
It might be good to described the reason of extending RelRule that RelOptRule will be deprecated eventually in Calcite 2.0 and we should start to replace RelOptRule with RelRule. This will help the reviewers get the context, since there are still many RelRuleOpt*** related code used.

Reply:
added in PR description